### PR TITLE
fix encrypt bug, add Event.quick

### PR DESCRIPTION
**Pull request by ntheden**
Fix a bug in the decipher, add Event.quick because it's useful, and add send and receive unit tests to start with.

### CODE REVIEW

File: lib/src/nips/nip_004/event.dart [hunk 1]
```diff
@@ -45,8 +45,24 @@ class EncryptedDirectMessage extends Event {
     return ciphertext;
   }
 
-  String getPlaintext(String receiverPrivkey, String senderPubkey) {
-    return Nip4.decipher(receiverPrivkey, senderPubkey, content);
+  String getPlaintext(String receiverPrivkey, [String senderPubkey=""]) {
```
Comment (ntheden) [inline lib/src/nips/nip_004/event.dart:48]: Decrypting an event should always use  the sender pub key provided in the event (not in the tags). But I left `senderPubkey` as an optional parameter.

Other thing is, if the decryption fails, there is no way to indicate that to the caller. I think `getPlaintext()` should throw an exception when decryption fails. @ethicnology do you have an opinion on that?

File: lib/src/event.dart [hunk 2]
```diff
@@ -271,6 +271,20 @@ class Event {
     );
   }
 
+  factory Event.quick(
```
Comment (no-prob) [inline lib/src/event.dart:274]: I find this convenient, can remove if not wanted.

File: lib/src/nips/nip_004/event.dart (hunk 1 above)
Comment (ethicnology) [reply lib/src/nips/nip_004/event.dart:48]: I did this refactoring based on the signature of your function decrypt but you are right, this should be as simple as possible so we should  probably remove this param

File: lib/src/event.dart (hunk 2 above)
Comment (ethicnology) [inline lib/src/event.dart:274]: I don't think I will keep `Event.quick` to keep as less code as possible in the library.

The `quicky` way to create a basic event is to use `Event.from` which only require the `kind`, `tags`, `content` and `privkey` supposing that everyone want to create a `kind=1` event is a huge assumption while in the context of nip4 we reduced the possibilities.

I can initialize `tags` with a empty array to reduce even more the input params to `kind` `content` and `privkey`

**Review by ethicnology — COMMENTED** [review]
@no-prob Thanks for this patch

I'm glad to work with you on it.

I can let you change what we talked about or i can do it myself (tell me)

https://github.com/ethicnology/dart-nostr/commit/385f211a4839eaf72cb67400352a290ec361dfa3?notification_referrer_id=NT_kwDOAYjwnbM2MTg5MzcxMzMyOjI1NzUxNzA5#commitcomment-109726063

File: lib/src/event.dart (hunk 2 above)
Comment (ntheden) [reply lib/src/event.dart:274]: Ok

**Comment by ntheden** [comment]
> I can let you change what we talked about or i can do it myself (tell me)

I will write the rest of the mentioned test cases now, the ones mentioned in
https://github.com/ethicnology/dart-nostr/issues/15#issuecomment-1511246032

But I wasn't sure what you meant with this question:
> Could generates random nostr keys to generate unit tests for Encrypted Direct Message with the hardcoded keys ?

**Pull request merged**